plugins/inputs/t128_lte/t128_lte.go

```diff
@@ -240,44 +225,38 @@ func readT128Config(t128ConfigFile string) ([]byte, error) {
 	return content, nil
 }
 
-func matchNode(nodeName string, content []byte) (string, error) {
-	var nodeData nodes
+func matchNode(nodeName string, content []byte) (t128Config, error) {
+	var nodeData t128Config
 	err := json.Unmarshal(content, &nodeData)
 	if err != nil {
-		return "", fmt.Errorf("Cannot unmarshal %s : %w", content, err)
+		return t128Config{}, fmt.Errorf("Cannot unmarshal %s : %w", content, err)
 	}
 	for _, router := range nodeData.Config.Authority.Router {
 		for _, name := range router.Node {
 			if name.Name == nodeName {
-				return nodeName, nil
+				return nodeData, nil
```

Comment:
This function is still not doing anything useful. It does verify that the expected node exists in the config, but that can be assumed. It should identify and return the data relevant for the current node. What you really want it to do is:
```go
type t128Config struct {
	Config struct {
		Authority struct {
			Router []struct {
				Name string `json:"name"`
				Node []nodeConfig `json:"node"`
			} `json:"router"`
		} `json:"authority"`
	} `json:"config"`
}

type nodeConfig struct {
    Name            string `json:"name"`
    DeviceInterface []lteDeviceInterface
}

func matchNode(nodeName string, content []byte) (nodeConfig, error) {
	// ...
}

func (plugin *T128Lte) getDeviceInterfaces(cfg nodeConfig) ([]lteDeviceInterface, error) {
	// no more need to check for node because why would we want to do that twice?
}
```

Reply:
You are still returning the `t128Config` from this function instead of the `nodeConfig` I described. It's not really a big deal in the scheme of things, but you're redoing work in `getDeviceInterfaces` that's already been done here. Are you not understanding my ask or is there another reason this change keeps not happening as I've described?

Reply:
Sorry I might have misunderstood! Fixed it and pushed the PR